core/src/engine/task/task_context.cpp

```diff
@@ -702,6 +709,26 @@ void TaskContext::TraceStateTransition(Task::State state) {
                       << logging::LogExtra::Stacktrace(logger);
 }
 
+void TaskContext::ResetPayload() {
+  if (!payload_) return;
+
+  std::exchange(payload_, nullptr)->~WrappedCallBase();
+}
+
+void intrusive_ptr_add_ref(TaskContext* p) noexcept {
+  p->intrusive_refcount_.fetch_add(1, std::memory_order_relaxed);
+}
+
+void intrusive_ptr_release(TaskContext* p) noexcept {
+  if (p->intrusive_refcount_.fetch_sub(1, std::memory_order_relaxed) == 1) {
```

Comment:
This is a very complicated question, what memory order to use here. Different C++ standard libraries implement this place differently.

Jus use seq_cst here and in intrusive_ptr_add_ref